async-http-client-backend/zio/src/test/scala/sttp/client/asynchttpclient/zio/SttpBackendStubZioTests.scala

```diff
@@ -26,4 +28,29 @@ class SttpBackendStubZioTests extends AnyFlatSpec with Matchers with ScalaFuture
     runtime.unsafeRun(r).body shouldBe Right("c")
     runtime.unsafeRun(r).body shouldBe Right("a")
   }
+
+  it should "return given web socket response" in {
+    val monad = new RIOMonadAsyncError[Any]
+    val frame1 = WebSocketFrame.text("initial frame")
+    val sentFrame = WebSocketFrame.text("sent frame")
+    val webSocketStub =
+      WebSocketStub
+        .respondWithMessages(List(frame1))(monad)
```

Comment:
would it be possible not to require the `monad` instance here? I.e. make `WebSocketStub` "lazy", parametrised with the monad instance, which would be given later by the `SttpBackendStub` (which already knows it). That would make this more user-friendly :)

Reply:
The price is either loosing power to set "failed response" or requiring monad instance when setting responses (which doesn't make it better).

Reply:
hm can you give an example of setting a failed response? What would be the use-case?

Reply:
`.respondUsingF(_ => List(monad.error(new Exception("boom!")), monad.unit(Right(frame1))))`
Would err on the first `receive` and succeed on the second.

Reply:
hm but when you create a WebSocketStub, you already know the concrete effect type, so the user can just pass in the failed effect? Without having to provide the sttp monad

Reply:
`.respondUsingF(_ => List(RIO.fail(new Exception("boom!")), ZIO.succeed(Right(frame1))))`
Is valid as well.

Let's assume we don't have an instance of monad available when setting stub.
`type Message = Either[WebSocketEvent.Close, WebSocketFrame]`
Choices I see:
1. we store `List[Message]` and user have no power to set error
1. we store `List[F[Message]]` and user cannot use convenient API (for happy-paths) that takes frames without `F`, because we can't store it without `monad.unit`
1. we store `List[Try[Message]]` we have both APIs

I'll give the last option a shot

Reply:
4. we store `List[SttpMonad => F[Message]]`, which would allow us to provide the convenient API as well?

Reply:
I don't see much improvement over `Try` because I can simply translate `Success(a)` => `monad.unit(a)`, `Failure(t)` => `monad.error(t)`>
Would it be more convenient for users to supply `Message => List[Try[Message]]` or `Message => MonadError => List[F[Message]]`?

Still I think that for most complicated cases, users should fallback to implementing `WebSocket` themselves.

Reply:
ok, let's try `Try` :)